### ISSUE TITLE
If I put angular.module().factory() up top and then function definition below

### ISSUE THREAD

**Issue by kylegoetz** [johnpapa/angular-styleguide#359]
I get all kinds of wacky errors about undefined everything. But if I ignore the style guide and put the function definition up top, it all works. Styleguide should be updated accordingly.

What I mean is:

angular.module('myApp').factory('myFactory', myFactory);
function myFactory() { ... }

fails, but

function myFactory() { ... }
angular.module('myApp').factory('myFactory', myFactory);

succeeds. Very latest Firefox.

**Comment by gkalpak** [comment]
Considering that the former format works without issues for thousands of people, I guess the problem lies somewhere else (e.g. your specific setup).

**Comment by johnpapa** [comment]
What is telling you "all kinds of wacky errors about undefined everything" ? JSHint ? The browser console? JSCS? Something else? Please help us help you by telling us what is telling you the issue, and what the exact error message is.

One place to look is JSHint and JSCS, which if not configured similarly to the .jscsrc file in this guide, they may complain.

**Comment by kylegoetz** [comment]
It was the Javascript console in Firefox (latest version, it auto-updates, right?). Supposing "auth" is a service that has "foo" factory injected into it, I was getting the error fooProvider undefined "foo <- auth" or something like that. I literally switched the code around like I mentioned, and the error went away.

I'm totally confused, but it works now, so I'm not going to complain or try to figure it out right now any further. I had seen another style guide somewhere mention that you might have to write it the way I did (put function myFunc before the angular.module('myApp').factory('myServ', myFunc)) to avoid some kind of undefined function error, so I figured that was it and made my initial issue report just to bring attention to the style guide needing to be updated to reflect that fact.

I didn't really intend to try to debug the issue or anything! I'll keep this tab open in the background so when I get good progress with this project I can come back and provide more info if it might actually be an issue with the style guide! Thanks for the feedback, though!
